lib/services/local/presence.go

```diff
@@ -1577,6 +1580,38 @@ func (s *PresenceService) listResources(ctx context.Context, req proto.ListResou
 	}, nil
 }
 
+// UpsertDatabaseService creates or updates (by name) a DatabaseService resource.
+func (s *PresenceService) UpsertDatabaseService(ctx context.Context, service types.DatabaseService) (*types.KeepAlive, error) {
```

Comment:
Why is this method not in lib/services/local/databaseservice.go? Other database service backend methods are there.

Reply:
I tried to recall the reasoning on this one, but I don't recall it.

Well, I think this should be changed.
Given that this PR is a backport, I would rather open a new PR against master and backport it to v12 and v11

Unless, you think this might cause any issues when running.

Reply:
Up to you, new PR to move it sounds good to me too.

Reply:
Created here
https://github.com/gravitational/teleport/pull/20930

Reply:
The above PR was merged into `master` and I'm also including it here in this PR (cherry pick from `master`)